zulipterminal/ui_tools/boxes.py

```diff
@@ -500,6 +501,16 @@ def footer_notify_invalid_recipient(self, error_msg: str) -> None:
         ]
         self.view.set_footer_text(footer_message, 3)
 
+    def loosing_focus_from_stream_box(self, key: str) -> bool:
+        stream_header = self.header_write_box[self.FOCUS_HEADER_BOX_STREAM]
+        stream_name_len = len(stream_header.edit_text)
+        cursor_pos = stream_header.edit_pos
+        if (is_command_key('CYCLE_COMPOSE_FOCUS', key)
+                or key in ['right', 'down']):
+            if key != 'right' or stream_name_len == cursor_pos:
+                return True
```

Comment:
The nested conditional could be simplified with what follows.
```python3
        if (is_command_key('CYCLE_COMPOSE_FOCUS', key) or key == "down"
              or (key == "right" and stream_name_len == cursor_pos)):
            return True
 ```
 
 While writing conditionals, expressing the intent clearly is vital. For instance, in the snippet above it is comparatively apparent that we're checking whether the key is `TAB` or `DOWN` or `RIGHT` (given that the cursor is at the end) for blur ("focus loss") event. Both approaches work, but the latter is more comprehensible.

Reply:
Yep, this is definitely more clear and comprehensible than the earlier conditional. Guess I'll have to look from different angles if the conditionals are getting complex. Thanks. :+1: